other/upgrade.php

```diff
@@ -1186,7 +1186,7 @@ function UpgradeOptions()
 	// Add proxy settings.
 	if (!isset($GLOBALS['image_proxy_maxsize']))
 		$changes += array(
-			'image_proxy_secret' => '\'' . substr(sha1(mt_rand()), 0, 20) . '\'',
+			'image_proxy_secret' => '\'' . substr(sha1(random_int(0, PHP_INT_MAX)), 0, 20) . '\'',
 			'image_proxy_maxsize' => 5190,
```

Comment:
This change is not needed.

Reply:
Why here not? this once call and add some better numbers.

Reply:
This doesn't seem to be for security tho. @Sesquipedalian should add his input.

Reply:
It doesn't matter either way. The image_proxy_secret isn't cryptographic, it's just obfuscation, so mt_rand() would be good enough. But this call is run once and never again, so there's no significant penalty to pay for using a CPRNG here. So unless this change causes errors in the upgrade process on older versions of PHP, let's keep it.